python/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+
+IF(NOT Kokkos_ENABLE_PYBIND11)
+    return()
+ENDIF()
+
+
+# don't build a static python module
+SET(BUILD_SHARED_LIBS ON)
+
+# pybind11 has not migrated to CMAKE_CXX_STANDARD and neither has kokkos
+if(NOT Kokkos_CXX_STANDARD)
+    set(Kokkos_CXX_STANDARD 11)
+endif()
+IF(MSVC)
+    set(PYBIND11_CPP_STANDARD /std:c++${Kokkos_CXX_STANDARD} CACHE STRING "C++ Language standard used by Pybind11")
+ELSE()
+    set(PYBIND11_CPP_STANDARD -std=c++${Kokkos_CXX_STANDARD} CACHE STRING "C++ Language standard used by Pybind11")
+ENDIF()
+
+# basically just used to get Python3_SITEARCH for installation
+FIND_PACKAGE(Python3 REQUIRED COMPONENTS Interpreter Development)
+# python binding library
+FIND_PACKAGE(pybind11 REQUIRED)
+
+pybind11_add_module(libpykokkos MODULE THIN_LTO
+    ${CMAKE_CURRENT_LIST_DIR}/libpykokkos.cpp)
+
+# link to kokkos
+TARGET_LINK_LIBRARIES(libpykokkos PRIVATE kokkoscore)
+
+# set the output path to <BINARY_DIR>/kokkos so one
+# can test the python import from the build directory
+# Really, only LIBRARY_* is needed for Unix but Windows
+# builds are weird so just setting all of them
+SET_TARGET_PROPERTIES(libpykokkos PROPERTIES
+    LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/kokkos
+    ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/kokkos
```

Comment:
Careful with `PROJECT_BINARY_DIR`. I don't totally understand what you're trying to do, but this will be different if someone is building Trilinos.

Reply:
This sets it up so that from the build directory, you can do `python -c "import kokkos; ..."` (or launch interactive session) without having to install kokkos

Reply:
It is mainly for testing purposes